src/Microsoft.Extensions.Configuration.FileProviderExtensions/FileConfigurationProvider.cs

```diff
@@ -0,0 +1,140 @@
+            : this(path, optional: false, reloadOnFileChanged: false)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="FileConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="path">Absolute path of the configuration file.</param>
+        /// <param name="optional">Determines if the configuration is optional.</param>
+        /// <param name="reloadOnFileChanged">Determines if the configuration provider should be reloaded if the file changes.</param>
+        public FileConfigurationProvider(string path, bool optional, bool reloadOnFileChanged)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentException("File path must be a non - empty string.", nameof(path));
+            }
+
+            Optional = optional;
+            Path = path;
+            ReloadOnFileChanged = reloadOnFileChanged;
+        }
+
+        /// <summary>
+        /// Gets a value that determines if this instance of <see cref="FileConfigurationProvider"/> is optional.
+        /// </summary>
+        public bool Optional { get; }
+
+        /// <summary>
+        /// The absolute path of the file backing this instance of <see cref="FileConfigurationProvider"/>.
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// If true, the configuration will be reloaded if the file changes.
+        /// </summary>
+        public bool ReloadOnFileChanged { get; }
+
+        /// <summary>
+        /// Called when the IConfigurationRoot is being built.
+        /// </summary>
+        /// <param name="root"></param>
+        public override void Initialize(IConfigurationRoot root)
+        {
+            if (ReloadOnFileChanged)
+            {
+                ReloadOnFileChange();
+            }
+        }
+
+        /// <summary>
+        /// Loads the contents of the file at <see cref="Path"/>.
+        /// </summary>
+        /// <exception cref="FileNotFoundException">If <see cref="Optional"/> is <c>false</c> and a
+        /// file does not exist at <see cref="Path"/>.</exception>
+        public override void Load()
+        {
+            if (!File.Exists(Path))
+            {
+                if (Optional)
+                {
+                    Data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                }
+                else
+                {
+                    throw new FileNotFoundException($"The configuration file '{Path}' was not found and is not optional.");
+                }
+            }
+            else
+            {
+                using (var stream = new FileStream(Path, FileMode.Open, FileAccess.Read))
+                {
+                    Load(stream);
+                }
+            }
+        }
+
+        public abstract void Load(Stream stream);
+
+        public IFileProvider FileProvider { get; set; }
```

Comment:
I am not sure it is needed. If it isn't then I would like to remove it. If it is needed then I think we should rearrange things to that it gets set in product code so that if anyone needs to call the getter it won't always be null.